### DISCUSSION TITLE
有没有兴趣试试带辅助码的双拼输入法呢？

### DISCUSSION THREAD

**Discussion by functoreality** [oldwinter/knowledge-garden#14] [分享]
先赞一个你公开的数字笔记。要展示双链的作用，实际的例子比起纯粹说理式文章确实能提供更切身的感受，显得直观很多。
最近是在笔记库里看到了 [搜狗输入法](https://oldwinter.top/Cards/%E6%B0%B8%E4%B9%85%E7%AC%94%E8%AE%B0/%E6%90%9C%E7%8B%97%E8%BE%93%E5%85%A5%E6%B3%95) 这篇笔记，于是来宣传一下自己做的 [Rime 双拼加辅助码输入项目](https://github.com/functoreality/rime-flypy-zrmfast)。可以当纯双拼输入法来用，此外可以根据字的偏旁，补充输入辅助码（或者叫形码）来进一步筛选候选项，达到减少翻页的目的。（我一页只排 4 个候选，好像很久没有在敲中文的时候翻页了，即使是输入各种人名。）
词库迁移问题确实存在，不过个人认为在辅助码加持下问题相对来说没那么大——毕竟所有词库里没有的词组都可以通过辅助码迅速造出来。我当初从搜狗迁移过来的时候完全没做词库迁移（知道如何迁移，只是觉得没必要），现在多设备使用也没做词库同步（同样没感到必要性，自己只做了配置的同步）。
另外，项目所依托的 Rime 输入法是开源软件，免费、无广告、隐私安全（甚至没联网功能）是自然的，也不怕像私有软件那样有后续无人维护的问题。还有一个许多开源软件有的特点，配置没有 GUI，都是 YAML 文件，这对一些人可能不那么方便，不过对于作者你这样的程序员来说或许不一定是缺点😂

最后再带一个私货。我最近也试图输出一些个人关于笔记系统的观点，目前只有[MOC的组织](https://functoreality.github.io/blog-pkm/contents/MOC%E7%9A%84%E7%BB%84%E7%BB%87/)这篇文章总体上写完了。也算是初步探索了一下嵌套展开的这种文章呈现方式。期待能够多多交流~

**Comment by oldwinter** [comment]
赞。现在这块儿同行者还很少，必须要关注一手啊。你页面下边的github地址点进去404了。

辅助码我之前尝试过小鹤形码几个月的，没坚持住，虽然不用选词很爽，能让我们盲打了。但我一方面不太在意偶尔出现的错别字，另一方面极度需要云选词的功能，比如人生若只如初见啦，何事秋风悲画扇啦等等词语用双拼很容易打出来，但形码辅助码方案就做不到了。

综合之下我后来就放弃了，但一定有些职业和性格的人，更需要辅助码~~

你界面里折叠展开功能怎么做的？用到了html嘛？

**Reply by functoreality** [reply]
古诗词这种比较长的固定短语没有云选词确实不太方便。虽然我的这个输入法方案里可以添加古诗词词典，不过现在用的词典里居然没你说的这两句，不知道社区之后会不会有人改进一下。
说起来输入这种长短语可能纯声母就够了，所以我的古诗词词典都是挂在全拼下的，然后双拼里是用 `ap` 前缀临时进入全拼模式。搜狗好像没为双拼设计这种临时全拼输入的功能，如果有的话或许还会再方便一些。

我页面下的地址应该是[这个](https://github.com/functoreality/blog-pkm)（404 是因为有个 `/` 被自动替换成 `%2f` 了），确实用了 HTML+CSS 的语法。折叠展开这个功能应该不算新，很多大纲笔记都支持，只是默认状态好像一般是全部展开，而不是全部折叠。

**Comment by kuifye** [comment]
我在尝试小鹤双拼+五笔形码的直接辅助码编码模式，用了一段时间感觉还可以。
